sklearn/decomposition/dict_learning.py

```diff
@@ -94,7 +101,7 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         cov = np.dot(dictionary, X.T)
 
     if algorithm == 'lasso_lars':
-        alpha = float(regularization) / n_features  # account for scaling
+        alpha = float(regularization) / sqrt(n_features)  # account for scaling
```

Comment:
This scaling is the one recommended in the Lasso litterature. I can write a test showing that sparsity elvel stays roughly the same no matter what input feature size is provided.